source/commands/nick.cpp

```diff
@@ -1,9 +1,38 @@
 #include "commands.hpp"
 
+bool checkNick(std::string nick)
+{
+	if (!((nick[0] >= 'a' && nick[0] <= 'z') || (nick[0] >= 'A' && nick[0] <= 'Z')))
+		return (false);
+	return (true);
+}
+
 void cmd::nick(const ft::Message& msg, ft::Client& client, ft::IRC& irc)
 {
 	(void)irc;
+	if (!client._raspberry)
+	{
+		//client.sendmsg(ft::Message(":127.0.0.1 462 :Unauthorized command (already registered)")); //ERR_ALREADYREGISTRED
+		return ;
+	}
+	if (msg.parameters.size() != 1)
+	{
+		client.sendmsg(ft::Message(":127.0.0.1 431 :No nickname given")); //ERR_NONICKNAMEGIVEN
+		return ;
+	}
+	if (msg.parameters.at(0).empty() || !checkNick(msg.parameters.at(0)))
+	{
+		client.sendmsg(ft::Message(":127.0.0.1 432 :Erroneus nickname")); //ERR_ERRONEUSNICKNAME
+		return ;
+	}
+	if (client.getNick().empty() == false)//CHECK
+	{
+		client.sendmsg(ft::Message(":127.0.0.1 433 :Nickname is already in use")); //ERR_NICKNAMEINUSE
+		return ;
+	}
+	// ERR_NICKCOLLISION 
```

Comment:
I think `ERR_NICKNAMEINUSE` and `ERR_NICKCOLLISION` have the same effect for us. Normally the `ERR_NICKNAMEINUSE` would be returned if there was a nickcollision happening on our server, while `ERR_NICKCOLLISION` would be returned if there was a nickcollision on the whole network. But since we only have one server, I'd suggest only returning `ERR_NICKNAMEINUSE`. See suggestion for further details
```suggestion
	if (irc._connections.find(msg.parameters.at(0)) != irc._connections.end())
	{
		// replacement for network-wide ERR_NICKCOLLISION 
		client.sendmsg(ft::Message(":127.0.0.1 433 :Nickname is already in use")); //ERR_NICKNAMEINUSE
		return ;
	}
```